app/shared/session/useLoginErrors.ts

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import { sentryException } from "../sentry";
+
+export const AUTH_PROVIDERS = {
+  LOGIN_DEV: "login_dev",
+  LOGIN_GITHUB: "login_github",
+  LOGIN_GOOGLE: "login_google",
+};
+
+export const LOGIN_ERROR_MESSAGES = {
+  [AUTH_PROVIDERS.LOGIN_DEV]: "There has been an issue logging you in with dev",
+  [AUTH_PROVIDERS.LOGIN_GITHUB]:
+    "There has been an issue logging you in with Github",
+  [AUTH_PROVIDERS.LOGIN_GOOGLE]:
+    "There has been an issue logging you in with Google",
+};
+
+export const useLoginErrors = (): string => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [messageToReturn, setMessageToReturn] = useState("");
+  const error = searchParams.get("error");
```

Comment:
error and message are only used inside the useEffect hook, so they could be defined there

Reply:
Done!